texastribune/queso-tools
# Build improvements

#### What's this PR do?

- Adds a new [CSS processor](https://github.com/parcel-bundler/lightningcss)
- Bumps many dependencies to latest version (except for ora)
- Drops AMP task runner
- Introduces requirement of node 14+

#### Why are we doing this? How does it help us?

Faster builds, security, smaller CSS output, cleans up tech debt

#### How should this be manually tested?

`npm run test` (kicked off in Github actions) should do the trick! I tested this in texastribune repo and did not see any breaking styles.

#### Are there any smells or added technical debt to note?

The implementation of the new inline source maps are a little ick, but after lots of research of how to do that with Lightning CSS and dart-sass, I'm just happy it works.

**Research links:**
- https://pub.dev/documentation/sass/latest/sass/sass-library.html
- https://11ty.rocks/posts/process-css-with-lightningcss/
- https://github.com/sass/dart-sass/issues/1594#issuecomment-1013208452
- https://github.com/parcel-bundler/lightningcss/blob/c34af42a333f784bd733bbd59a99651fbce351c5/website/pages/docs.md?plain=1#L173-L179
